### PR TITLE
refactor: add empty data structures and tests for phase information

### PR DESCRIPTION
**Pull request by catsmanac**
Add production and consumption records for phases to envoy data model.
Extend common properties with phase setup information.
Add tests for empty data structures added.
Update snapshot with empty data structures for production and consumption phases.
New structures are not used yet, they all contain their default values.

### CODE REVIEW

**Comment by catsmanac** [comment]
Anything I should do now to move this forward?

**Comment by bdraco** [comment]
> Anything I should do now to move this forward?

I missed this one with all the github spam.  Feel free to ask in our discord channel if something sits for a long time.

**Review by bdraco — APPROVED** [review]
I missed tih

File: tests/test_endpoints.py [hunk 1]
```diff
@@ -1937,3 +2114,20 @@ async def test_with_7_x_firmware(
...
+
+    if production_phases is None:
+        assert data.system_production_phases is None
+    if consumption_phases is None:
+        assert data.system_consumption_phases is None
+
+    if data.system_production_phases is None:
+        assert not production_phases
+    if envoy.data.system_consumption_phases is None:
+        assert not consumption_phases
```
Comment (bdraco) [inline tests/test_endpoints.py:2133]: We usually don't want branching in tests as it makes them harder to maintain

Comment (catsmanac) [reply tests/test_endpoints.py:2133]: I see. But I need to deal with some condtional situations, much like tests considering supported_features. Tried to come up with some logic for the assert to avoid the if but it would make the assert condition way more complex as the if statement. For now I can remove them as there's no data yet and I'll give it another shot with the data populated.

Comment (bdraco) [reply tests/test_endpoints.py:2133]: Few options here.

1. Remove the test (probably best choice for now)
2. Use pytest.mark.parametrize
3. Split the test into separate tests

Comment (catsmanac) [reply tests/test_endpoints.py]: For now I've removed them.

**Review by bdraco — APPROVED** [review]
